src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction.java

```diff
@@ -342,7 +343,17 @@ private static void adjustProjectDisplayName(
     private static Set<Run<?, ?>> getDownstreamBuilds(
             final Run build) {
 
-        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>();
+        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>(new Comparator<Run<?, ?>>() {
+            @Override
+            public int compare(Run<?, ?> o1, Run<?, ?> o2) {
+                final int res = o1.getParent().getFullName().compareTo(o2.getParent().getFullName());
```

Comment:
@rsandell I see you point, but according Jenkins Core source code:
```java
/**
 * The project this build is for.
 */ 
public @Nonnull JobT getParent() {
    return project;
}
```
So, `@Nonnull` says that `getParent()` can not return null. Should I add check in this case? And mark `getParent()` as `@Nullable` in Jenkins Core?

> What matters to me is making sure there are no risk for NullPointerExceptions

There is a check https://github.com/jenkinsci/build-failure-analyzer-plugin/pull/65/commits/a5f5fbea7445ccec24a0c52b24b97ce9e9fe26df#diff-d0bff283a6b9423a6895c868f62a3492R363. So, if we don't have `null`s in TreeSet, if we can't put `null`, because we have check for it, and if `getParent` never returns `null` we are safe. I can add only one small thing - return `Collections.unmodifiableSet(foundDbf)` instead of `foundDbf`, if you want. But right now there is no other attempts to modify this set in BFA source code.